mobile/src/main/java/org/openhab/habdroid/ui/activity/WebViewFragment.kt

```diff
@@ -163,7 +162,11 @@ class WebViewFragment : Fragment(), ConnectionFactory.UpdateListener {
 
     fun goBack(): Boolean {
         if (webView?.canGoBack() == true) {
+            val oldUrl = webView?.url
             webView?.goBack()
+            if (oldUrl == webView?.url) {
+                goBack()
```

Comment:
Isn't this equivalent to returning `webView?.url != oldUrl`? I'd find this more elegant than the recursive call.

Reply:
No, it's not the same. If `goBack()` returns `false`, HABPanel is closed.
I added this code to "filter out" redundant entries in the back stack.

Reply:
I see. The recursion still irks me a bit. How about this:
```
val oldUrl = webView?.url
do {
    webView?.goBack()
} while (webView?. url == oldUrl && webView?.canGoBack() == true)
```

And maybe add a comment like `// skip redundant history entries while going back`

Reply:
Picked.